### PR TITLE
Manacher's algorithms - in Python 

### PR DESCRIPTION
**Pull request by Sean16SYSU**
- [X] **If creating a new file :**
  - [X] added links to it in the README files ?
  - [X] included tests with it ?
  - [X] added description (overview of algorithm, time and space complexity, and possible edge case) in docstrings ?

### CODE REVIEW

**Comment by Sean16SYSU** [comment]
I have already fixed the conflicts.

File: algorithms/strings/longest_palindromic_substring.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+Example2:
+
+* input: "cbbd"
+* output: "bb"
+
+Manacher's algorithm
+
+'''
+
+def longestPalindrome(s):
```
Comment (ericklarac) [inline algorithms/strings/longest_palindromic_substring.py:18]: Convert to snake case (```longest_palindrome```), please.

Comment (Sean16SYSU) [reply algorithms/strings/longest_palindromic_substring.py:18]: Ok, thanks for your suggestion, and I have already fixed the case with using much more complex testing case.

Comment (ericklarac) [reply algorithms/strings/longest_palindromic_substring.py:18]: I was suggesting to change the name of the function from came case (```longestPalindrome```) to snake case (```longest_palindrome```), to match the format from the other files.

Comment (Sean16SYSU) [reply algorithms/strings/longest_palindromic_substring.py:18]: ooo, lol, get it.
